source/extensions/transport_sockets/tls/context_impl.cc

```diff
@@ -1047,6 +1094,29 @@ ServerContextImpl::ServerContextImpl(Stats::Scope& scope,
     int rc =
         SSL_CTX_set_session_id_context(ctx.ssl_ctx_.get(), session_id.data(), session_id.size());
     RELEASE_ASSERT(rc == 1, Utility::getLastCryptoError().value_or(""));
+
+    auto& ocsp_resp_bytes = tls_certificates[i].get().ocspStaple();
+    if (ocsp_resp_bytes.empty()) {
+      if (Runtime::runtimeFeatureEnabled(
+              "envoy.reloadable_features.require_ocsp_response_for_must_staple_certs") &&
```

Comment:
Why is this a runtime feature, not config?

Reply:
If your TLS certificate has the must-staple extension and you don't have a valid OCSP response to staple, I think you're generally not going to be able to form new TLS sessions (though implementations likely vary). I don't think it's a generally useful configuration option, since over time I would expect more and more clients to correctly reject the TLS handshake. We added this on the off chance that some operator has specific knowledge that their configuration will work with the invalid response (e.g. they control the client, or know that most of their users run a client that ignores the error) and wants to allow it, presumably while they repair their OCSP stapling infrastructure.